### PR TITLE
Bug 1202423 - Add default app order configuration. r=gmarty,kgrandon

### PR DESCRIPTION
**Pull request by Cwiiis**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=bb5715c109916df7a5e2a75c7c612c3e3b2a299f

File: apps/homescreen/build/default-homescreens.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      ["hosted_apps", "bugzillalite"],
+      ["hosted_apps", "facebook"],
+      ["hosted_apps", "twitter"],
+      ["apps", "costcontrol"]
+    ]
+  ],
+  "preferences": {
+    "grid.cols": 3
+  }
+}
```
Comment (gmarty) [inline apps/homescreen/build/default-homescreens.json]: Why copy the file from verticalhomescreen and not instead create a file with just the data needed? Is it to make migration easier?

File: apps/homescreen/js/app.js [hunk 2]
```diff
@@ -286,7 +295,30 @@ const SETTINGS_VERSION = 0;
...
+              populateApps();
+            });
+          },
+          (e) => {
+            console.error('Failed to load first-run script');
+            populateApps();
+          });
+      }
+
+      return populateApps();
```
Comment (gmarty) [inline apps/homescreen/js/app.js]: Any reason to `return`  here?

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9e53b16fddc291491e5f95a5e9dd020ad3d68cc4

File: apps/homescreen/build/default-homescreens.json (hunk 1 above)
Comment (Cwiiis) [inline apps/homescreen/build/default-homescreens.json]: No good reason, but I wanted the file to look familiar to whoever's responsible for updating it - I should perhaps remove the collections section, but it's nice to keep it there to test that it won't fall over if it encounters something it doesn't handle.

File: apps/homescreen/js/app.js (hunk 2 above)
Comment (Cwiiis) [inline apps/homescreen/js/app.js]: Just that populateApps is a promise, so if we ever wanted to chain in the future, we wouldn't spend ages debugging why it seems to execute immediately (something I ran into elsewhere :))

File: apps/homescreen/js/app.js [hunk 3]
```diff
@@ -286,7 +295,30 @@ const SETTINGS_VERSION = 0;
         console.error('Failed to initialise metadata db', e);
         resolve();
       });
-    }).then(populateApps);
+    }).then(() => {
+      if (this.firstRun) {
+        return LazyLoader.load(['js/firstrun.js'],
```
Comment (KevinGrandon) [inline apps/homescreen/js/app.js]: LazyLoader.load returns a promise which might make this more clean.

File: apps/homescreen/build/default-homescreens.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{
+  "homescreens": [
+    [
+      ["apps", "communications", "dialer"],
+      ["apps", "sms"],
+      ["apps", "communications", "contacts"],
+      ["apps", "email"],
+      ["apps", "search"]
+    ], [
+      ["apps/collection/collections", "social"],
```
Comment (KevinGrandon) [inline apps/homescreen/build/default-homescreens.json]: Remove these if we're not going to have collections.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6bd1412237488b28167afc7fc79e477df45ad686

File: apps/homescreen/js/app.js (hunk 3 above)
Comment (Cwiiis) [inline apps/homescreen/js/app.js]: Used the return result directly with a ternary. Marginally cleaner :)

File: apps/homescreen/build/default-homescreens.json (hunk 4 above)
Comment (Cwiiis) [inline apps/homescreen/build/default-homescreens.json]: Done.

File: apps/homescreen/build/default-homescreens.json (hunk 1 above)
Comment (Cwiiis) [inline apps/homescreen/build/default-homescreens.json]: As you and Kevin both flagged this, I removed the collections.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=fc5c1a0935368eec1a9f3e0ffd24a1bd71ffea97

**Pull request merged**